evalml/tests/objective_tests/test_objectives.py

```diff
@@ -194,60 +194,40 @@ def test_get_objectives_all_expected_ranges(obj):
 @pytest.mark.parametrize(
-    "nullable_ltype",
+    "nullable_y_true_ltype",
     ["BooleanNullable", "IntegerNullable", "AgeNullable"],
 )
-@pytest.mark.parametrize("y_bool_null_incompatible", [True, False])
-@pytest.mark.parametrize("y_int_null_incompatible", [True, False])
-@pytest.mark.parametrize("data_type", ["ww", "pd"])
-def test_objective_base_handle_nullable_types(
-    split_nullable_logical_types_by_compatibility,
+def test_objectives_support_nullable_types(
+    nullable_y_true_ltype,
+    obj,
     nullable_type_target,
-    data_type,
-    y_bool_null_incompatible,
-    y_int_null_incompatible,
-    nullable_ltype,
 ):
-    y_true = nullable_type_target(ltype=nullable_ltype, has_nans=False)
-    if data_type == "pd":
-        # remove woodwork types but keep nullable pandas dtype
-        y_true = y_true.copy()
-
-    obj = CustomObjective(
-        integer_nullable_incompatible=y_int_null_incompatible,
-        boolean_nullable_incompatible=y_bool_null_incompatible,
-    )
-    y_true_d = obj._handle_nullable_types(y_true)
-
-    (
-        y_compatible_ltypes,
-        y_incompatible_ltypes,
-    ) = split_nullable_logical_types_by_compatibility(
-        y_int_null_incompatible,
-        y_bool_null_incompatible,
-    )
+    y_true = nullable_type_target(ltype=nullable_y_true_ltype, has_nans=False)
+    y_pred = pd.Series([0, 1, 0, 1, 1] * 4, dtype="int64")
+    X = None
 
-    # pd data type will get non nullable logical type when woodwork is initialized, so we don't expect it to be the same
-    if data_type != "pd" and nullable_ltype in {
-        str(ltype) for ltype in y_compatible_ltypes
-    }:
-        assert isinstance(
-            y_true_d.ww.logical_type,
-            tuple(y_compatible_ltypes),
+    # Instantiate objective with any needed parameters
+    if obj == CostBenefitMatrix:
+        obj = obj(
+            true_positive=10,
+            true_negative=-1,
+            false_positive=-7,
+            false_negative=-2,
         )
     else:
-        assert not isinstance(
-            y_true_d.ww.logical_type,
-            tuple(y_incompatible_ltypes),
-        )
-
-
-def test_binary_objective_handle_nullable_types_numpy_input():
-    y_true = pd.Series([1, 0, 1, 1, 0] * 5)
-    y_true = y_true.to_numpy()
-
-    obj = CustomObjective()
-    y_true_d = obj._handle_nullable_types(y_true)
-
-    assert y_true_d is y_true
+        obj = obj()
+
+    # Make any changes to data needed for objective
+    if isinstance(obj, FraudCost):
+        # FraudCost needs an "amount" column
+        X = pd.DataFrame({"amount": [100, 5, 250, 89] * 5})
+    elif isinstance(obj, MAPE):
+        if isinstance(y_true.ww.logical_type, BooleanNullable):
+            pytest.skip("MAPE doesn't support inputs containing 0")
+        # Replace numeric inputs containing 0
+        y_true = y_true.ww.replace({0: 10})
+        y_pred = y_pred.replace({0: 10})
+
+    obj.score(y_true=y_true, y_predicted=y_pred, X=X)
```

Comment:
For sanity's sake, should we be checking any equality or non-nullness here?

Reply:
added a check that the value isnt null: b5ba0a2b9